chain/epoch_manager/src/tests/mod.rs

```diff
@@ -2442,3 +2451,179 @@ fn test_chunk_producers() {
+    let accounts = vec![
+        ("test0".parse().unwrap(), 1000),
+        ("test1".parse().unwrap(), 1000),
+        ("test2".parse().unwrap(), 1000),
+        ("test3".parse().unwrap(), 1000),
+        ("test4".parse().unwrap(), 500),
+    ];
+    let epoch_info = epoch_info(
+        0,
+        accounts,
+        vec![0, 1, 2, 3],
+        vec![vec![0, 1, 2], vec![0, 1, 3, 4]],
+        vec![],
+        vec![],
+        BTreeMap::new(),
+        vec![],
+        HashMap::new(),
+        0,
+    );
+    let (kickouts, validator_stats) = EpochManager::compute_kickout_info(
+        &epoch_config,
+        &epoch_info,
+        &HashMap::from([
+            (0, ValidatorStats { produced: 100, expected: 100 }),
+            (1, ValidatorStats { produced: 90, expected: 100 }),
+            (2, ValidatorStats { produced: 100, expected: 100 }),
+            // test3 will be kicked out
+            (3, ValidatorStats { produced: 89, expected: 100 }),
+        ]),
+        &HashMap::from([
+            (
+                0,
+                HashMap::from([
+                    (0, ValidatorStats { produced: 100, expected: 100 }),
+                    (1, ValidatorStats { produced: 80, expected: 100 }),
+                    (2, ValidatorStats { produced: 70, expected: 100 }),
+                ]),
+            ),
+            (
+                1,
+                HashMap::from([
+                    (0, ValidatorStats { produced: 70, expected: 100 }),
+                    (1, ValidatorStats { produced: 79, expected: 100 }),
+                    (3, ValidatorStats { produced: 100, expected: 100 }),
+                    (4, ValidatorStats { produced: 100, expected: 100 }),
+                ]),
+            ),
+        ]),
+        &HashMap::new(),
+        &HashMap::new(),
+    );
+    assert_eq!(
+        kickouts,
+        HashMap::from([
+            ("test1".parse().unwrap(), NotEnoughChunks { produced: 159, expected: 200 }),
+            ("test2".parse().unwrap(), NotEnoughChunks { produced: 70, expected: 100 }),
+            ("test3".parse().unwrap(), NotEnoughBlocks { produced: 89, expected: 100 }),
+        ])
+    );
+    assert_eq!(
+        validator_stats,
+        HashMap::from([
+            (
+                "test0".parse().unwrap(),
+                BlockChunkValidatorStats {
+                    block_stats: ValidatorStats { produced: 100, expected: 100 },
+                    chunk_stats: ValidatorStats { produced: 170, expected: 200 }
+                }
+            ),
+            (
+                "test4".parse().unwrap(),
+                BlockChunkValidatorStats {
+                    block_stats: ValidatorStats { produced: 0, expected: 0 },
+                    chunk_stats: ValidatorStats { produced: 100, expected: 100 }
+                }
+            ),
+        ])
+    );
+}
+
+#[cfg(feature = "protocol_feature_max_kickout_stake_ratio")]
```

Comment:
could you move this cfg somewhere 'inside' the test (so it is clearly visible who would have been kicked out with this feature vs without this feature)